source/agora/node/Registry.d

```diff
@@ -472,60 +472,102 @@ public class NameRegistry: NameRegistryAPI
     }
 }
 
+/*******************************************************************************
+
+    Parse a PublicKey from the first label of a domain name
+
+    Params:
+      domain = The full domain name to parse
+
+    Returns:
+      A valid `PublicKey`, or `PublicKey.init`
+
+*******************************************************************************/
+
+private PublicKey extractPublicKey (in char[] domain) @safe
```

Comment:
Regarding this change, since #2770 was using this as a function delegate in `TypedPayload.make`, that function can be changed to remove delegate parameter and directly calling this.

Reply:
Do we even call this function ? It would have stopped compiling if we did.

Reply:
Not right now but it will be called, I can do this in separate PR